src/containers/array.c

```diff
@@ -145,11 +145,7 @@ int array_container_shrink_to_fit(array_container_t *src) {
 
 /* Free memory. */
 void array_container_free(array_container_t *arr) {
-    if (arr->array !=
-        NULL) {  // Jon Strabala reports that some tools complain otherwise
-        roaring_free(arr->array);
-        arr->array = NULL;  // pedantic
-    }
+    roaring_free(arr->array);
```

Comment:
I believe we should still check that `arr != NULL` before accessing `arr->array`.

Reply:
added back the check.